presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestMultipleDistinctAggregationToMarkDistinct.java

```diff
@@ -0,0 +1,121 @@
+                                        p.symbol("input2")))))
+                .doesNotFire();
+    }
+
+    @Test
+    public void testSingleDistinct()
+    {
+        tester().assertThat(new MultipleDistinctAggregationToMarkDistinct())
+                .on(p -> p.aggregation(builder -> builder
+                        .globalGrouping()
+                        .addAggregation(p.symbol("output1"), expression("count(DISTINCT input1)"), ImmutableList.of(BIGINT))
+                        .source(
+                                p.values(
+                                        p.symbol("input1"),
+                                        p.symbol("input2")))))
+                .doesNotFire();
+    }
+
+    @Test
+    public void testMultipleAggregations()
+    {
+        tester().assertThat(new MultipleDistinctAggregationToMarkDistinct())
+                .on(p -> p.aggregation(builder -> builder
+                        .globalGrouping()
+                        .addAggregation(p.symbol("output1"), expression("count(DISTINCT input)"), ImmutableList.of(BIGINT))
+                        .addAggregation(p.symbol("output2"), expression("sum(DISTINCT input)"), ImmutableList.of(BIGINT))
+                        .source(
+                                p.values(p.symbol("input")))))
+                .doesNotFire();
+    }
+
+    @Test
+    public void testDistinctWithFilter()
+    {
+        tester().assertThat(new MultipleDistinctAggregationToMarkDistinct())
+                .on(p -> p.aggregation(builder -> builder
+                        .globalGrouping()
+                        .addAggregation(p.symbol("output1"), expression("count(DISTINCT input1) filter (where input2 > 0)"), ImmutableList.of(BIGINT))
+                        .addAggregation(p.symbol("output2"), expression("count(DISTINCT input2) filter (where input1 > 0)"), ImmutableList.of(BIGINT))
+                        .source(
+                                p.values(
+                                        p.symbol("input1"),
+                                        p.symbol("input2")))))
+                .doesNotFire();
+
+        tester().assertThat(new MultipleDistinctAggregationToMarkDistinct())
+                .on(p -> p.aggregation(builder -> builder
+                        .globalGrouping()
+                        .addAggregation(p.symbol("output1"), expression("count(DISTINCT input1) filter (where input2 > 0)"), ImmutableList.of(BIGINT))
+                        .addAggregation(p.symbol("output2"), expression("count(DISTINCT input2)"), ImmutableList.of(BIGINT))
+                        .source(
+                                p.values(
+                                        p.symbol("input1"),
+                                        p.symbol("input2")))))
+                .doesNotFire();
+    }
+
+    @Test
+    public void testMixedDistinctAndNonDistinct()
+    {
+//        tester().assertThat(new MultipleDistinctAggregationToMarkDistinct())
```

Comment:
please add TODO/comment why it's commented out

Reply:
I'll just remove it and put it in a github issue for later. I had it here for illustration and in case someone had ideas about something I might be missing.